Server/JokesOnYou.Web.Api/Services/JokesService.cs

```diff
@@ -10,10 +14,42 @@ namespace JokesOnYou.Web.Api.Services
     public class JokesService : IJokesService
     {
         private readonly IJokesRepository _jokesRepo;
+        private readonly IUserRepository _userRepository;
+        private readonly IUnitOfWork _unitOfWork;
+        private readonly IMapper _mapper;
 
-        public JokesService(IJokesRepository jokesRepo)
+        public JokesService(IJokesRepository jokesRepo, IUserRepository userRepository, IUnitOfWork unitOfWork, IMapper mapper)
         {
             _jokesRepo = jokesRepo;
+            _mapper = mapper;
+            _userRepository = userRepository;
+            _unitOfWork = unitOfWork;
+        }
+        public async Task<JokeReplyDto> CreateJokeAsync(JokeCreateDto jokeCreateDto, string userId)
+        {
+
+            var user = await _userRepository.GetUserAsync(userId);
+            var joke = _mapper.Map<Joke>(jokeCreateDto);
+            joke.UploadDate = DateTime.Now;
+            joke.Author = user;
+
+            await _jokesRepo.CreateJokeAsync(joke);
+            var saved = await _unitOfWork.SaveAsync();
+            if (!saved)
```

Comment:
Should this check be in place because some other method could have saved it before this gets called ?

Reply:
what other method could have saved this? Yeah service is the one handling the saving

Reply:
I thought because its Async another method somewhere else could save the Database before this one gets called ?
but if that's not possible then no check is needed.

Reply:
Ahh i get it, Yeah changes can be saved in other places. But isnt this check was to see that there were actually any changes saved and this save was successfull

Reply:
ill add a second check for when its not saved to see if its in the database ;)